src/TesterInternal/SerializerTests/CustomSerializerTests.cs

```diff
@@ -132,5 +133,13 @@ public void Serialize_GrainMethodAsyncReturnOnlyType()
         {
             Assert.IsNotNull(SerializationManager.GetSerializer(typeof(SerializerTestClass6)), "No serializer generated for return type of parameterless Task grain method");
         }
+        
+        [TestMethod, TestCategory("Functional"), TestCategory("Serialization")]
+        public void Serialize_AsyncObserverArgumentType()
+        {
+            var z = new ClosedTypeStreamObserver(TraceLogger.GetLogger("SerializationTests", TraceLogger.LoggerType.Runtime));
+
+            Assert.IsNotNull(SerializationManager.GetSerializer(typeof(AsyncObserverArg)), "No serializer generated for argument type of async observer");
```

Comment:
That would be rather sanity check and test of a generated serializer, but also having that basically does not hurt, so I've added additional assert.